src/definition/populate.go

```diff
@@ -116,7 +116,12 @@ func IntegrationPopulator(config *IntegrationPopulateConfig) (bool, []error) {
 				extraAttributes...,
 			)
 
-			msType, err := config.MsTypeGuesser(config.ClusterName, groupLabel, entityID, config.Groups)
+			msTypeGuesser := config.MsTypeGuesser
+			if customGuesser := config.Specs[groupLabel].MsTypeGuesser; customGuesser != nil {
+				msTypeGuesser = customGuesser
```

Comment:
Are we testing this overriding behaviour? Not separately but integrated.

Reply:
That is a good point 👍 
We are definitely testing it with the integration tests: if we didn't use it current _horizontalpodautoscaler_ wouldn't be found in `ksm_test`. However, I think it should be possible to unit-test this new behavior by extending the populator tests, so I am giving it a try. Thanks!

Reply:
I used the word `integrated` unfortunately here. I didn't mean in the integration tests but in the unit 😅

Reply:
I added a new populator test here to check the custom msTypeGuesser is actually used. [d50f349](https://github.com/newrelic/nri-kubernetes/pull/577/commits/d50f349b403eaa4af1b25178978d150004e13e72)